### ISSUE TITLE
Version of `f1^f2` that includes parent terms

### ISSUE THREAD

**Issue by grantmcdermott** [lrberge/fixest#61]
**tl;dr** Just as `f1^f2` essentially provides a fast alternative to `f1:f2` expansion in the FE slot, how about a fast alternative to `f1*f2` that includes parent terms (say, `f1^^f2`)?

Hi Laurent,

I've been working with some models that require interactions of fixed effects. Your `f1^f2` syntax is very nice and makes a demonstrable difference in speed when working with large datasets.

However, one minor downside is that &mdash; essentially being a fast paste operation &mdash; it doesn't include parent terms. So these must be specified manually, which leads to a fair bit of extra typing... at least, compared to the standard R expansion operation of `f1 * f2`.

What are your thoughts, then, about a version of `f1^f2` that does preserve parent terms (but still includes the speed benefits of the interaction term)? Maybe syntax of the form `f1^^f2`?

Again, this would really just be for convenience (i.e. less typing) so you may not want to prioritize.  It may also be a bit cumbersome to handle multiway interactions in same way that `f1 * f2 * f3` expands to `f1 + f2 + f3 + f1:f2 + f1:f3 + f2:f3 + f1:f2:f3` But I thought I'd raise it all the same!

**Comment by lrberge** [comment]
Hi Grant,

Maybe I misunderstood, but if you interact two variables you don't need the parents since they don't bring any extra information. The interaction is strictly nested in the parents: for example if you have state-year fixed-effects, then either the state FEs or the year FEs are strictly redundant.

Here's an illustration:
```R
base = iris
names(base) = c("y", "x1", "x2", "x3", "fe1")
base$fe2 = rep(letters[1:15], 10)

res = feols(y ~ x1 | fe1^fe2, base)
res_parent = feols(y ~ x1 | fe1^fe2 + fe1 + fe2, base)

rbind(coef(res), coef(res_parent))
#>             x1
#> [1,] 0.7447604
#> [2,] 0.7447604

summary(fixef(res_parent))
#> NOTE: The fixed-effects are not regular, they cannot be straightforwardly interpreted.
#> Fixed_effects coefficients
#>                         fe1^fe2 fe1 fe2
#> Number of fixed-effects      45   3  15
#> Number of references          0   3  15
#> Mean                       3.57   0   0
#> Standard-deviation        0.861   0   0
#> 
#> COEFFICIENTS:
#>   fe1^fe2: setosa_a setosa_b setosa_c setosa_d setosa_e                 
#>               2.493    2.575    2.293    2.519    2.431 ... 40 remaining
#> -----
#>   fe1: setosa versicolor virginica
#>             0          0         0
#> -----
#>   fe2: a b c d e                 
#>        0 0 0 0 0 ... 10 remaining
``` 

But maybe I missed something?

**Comment by grantmcdermott** [comment]
Oh god, you're right. Typing without thinking properly here. Sorry for distraction!

**Comment by lrberge** [comment]
No worries! ;-)
